autoblocks/_impl/tracer.py

```diff
@@ -101,6 +109,52 @@ def start_span(self):
                 props["span_id"] = prev_span_id
             self.set_properties(props)
 
+    async def _evaluate_event(self, event: TracerEvent, evaluator: BaseEventEvaluator) -> Optional[EventEvaluation]:
+        """
+        Evaluates an event using a provided evaluator.
+        """
+        evaluation = None
+        if inspect.iscoroutinefunction(evaluator.evaluate_event):
+            try:
+                evaluation = await evaluator.evaluate_event(event=event)
+            except Exception as err:
+                log.error("Event evaluation threw an exception", err)
+        else:
+            try:
+                ctx = contextvars.copy_context()
+                evaluation = await global_state.event_loop().run_in_executor(
+                    None,
+                    ctx.run,
+                    evaluator.evaluate_event,
+                    event,
+                )
+            except Exception as err:
+                log.error("Event evaluation threw an exception", err)
+
+        return evaluation
+
+    async def _run_evaluators(
+        self, evaluators: BaseEventEvaluator, event: TracerEvent, max_evaluator_concurrency: int
+    ) -> List[EventEvaluation]:
+        if len(evaluators) == 0:
+            return []
+        try:
+            evaluations: List[EventEvaluation] = await gather_with_max_concurrency(
+                max_evaluator_concurrency,
+                [
+                    self._evaluate_event(
+                        event=event,
+                        evaluator=evaluator,
+                    )
+                    for evaluator in evaluators
+                ],
+            )
+            return list(filter(lambda e: isinstance(e, EventEvaluation), evaluations))
+
+        except Exception as err:
+            log.error("Unable to complete evaluating event", err)
```

Comment:
<img width="492" alt="Screenshot 2024-02-27 at 9 10 32 AM" src="https://github.com/autoblocksai/python-sdk/assets/7498009/2d790d80-bb10-44b4-a00b-73ede93d3d6a">

the first argument should be the entire message. if you want to pass `err` separately like this you can use `%s` as a placeholder